### PR TITLE
add github ci

### PR DESCRIPTION
**Pull request by sparkoo**
Signed-off-by: Michal Vala <mvala@redhat.com>

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: Docker
```
Comment (benoitf) [inline .github/workflows/go.yml]: I would rename the file to build.yml as with go we don't really know what is being done

FYI it's missing copyright headers

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+on: [push]
+jobs:
+
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Build the Docker image
+      run: docker build . --file Dockerfile
```
Comment (benoitf) [inline .github/workflows/go.yml]: AFAIK `--file` is optional no ?

**Comment by benoitf** [comment]
Should the event be on `pull_request` ? because here you assume contribs are always pushed on this repository ?

**Comment by sparkoo** [comment]
> Should the event be on `pull_request` ? because here you assume contribs are always pushed on this repository ?

@benoitf yes, fixed that and your other comments. Thanks for the review

**Comment by davidfestal** [comment]
I'm not against it at all, but there is already a nightly build for the `che-operator` for every commit on master, on the centos CI afair. @ibuziuk do you confirm ?
And the idea was that at some we should extend this to also support PR flow.
Another point: I'm not sure that the general question about what CI server should be used (Centos or Eclipse CI) has been answered for now. @nickboldt @rhopp Any news about that ?

So what is exactly the purpose of using GitHub actions for this now ? Does this PR correspond to any opened and planned issue ?

**Comment by sparkoo** [comment]
@davidfestal it is not part of any planned issue. I'm currently working on few che-operator tasks and I miss CI feedback on pull request so I've added this simple check. When proper (whatever it means) CI will be set, I'm happy to remove this. But currently there is nothing.

**Comment by davidfestal** [comment]
I understand and will approve, though I don't see what it brings in addition to building the docker image locally, which I assume we already do when we develop.

**Pull request merged**